.github/workflows/publish_docs.yml

```diff
@@ -0,0 +1,14 @@
+name: Update Github Pages
+on:
+  push: { branches: master }
+jobs:
+  build_and_deploy:
+    name: Build & Deploy
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout main
+        uses: actions/checkout@v2
+      - name: Deploy MkDocs
+        uses: mhausenblas/mkdocs-deploy-gh-pages@1.24
```

Comment:
nice to see a version target on the image in use 👍🏼 

not blocking - fwiw i've been building these into our own org so we control the whole repo and don't pull in other code bases, e.g. i rebuilt the underlying dockerfile manually and copied deploy code into our image, https://github.com/zooniverse/middleman-gh-pages-action/blob/master/Dockerfile

it looks like `mhausenblas/mkdocs-deploy-gh-pages@1.24` relies on another base image https://github.com/mhausenblas/mkdocs-deploy-gh-pages/blob/1ef2f655b138f5a21bbee6344e03b25a4e2ba079/Dockerfile#L1 have we audited this? It's not locked to a known image tag so it'll use the latest image when rebuilt - these are supply chain attack vectors!

Looks like it's an apline python image which can build mkdocs with material theme.
https://github.com/squidfunk/mkdocs-material/blob/5462bf5b0c3df1bae7a281eaf7adf42ad6a15afc/Dockerfile

while i don't think this is a great concern with these images - it's worth considering locking these down vs relying on the open market.

Reply:
Ahhah! OK, I didn't properly consider the problem with supply chain attacks, thank you for spotting that massive weak point.

So your recommendation is that we host these GitHub actions on our side, correct? So I'll...

1. fork `github.com/mhausenblas/mkdocs-deploy-gh-pages` to `github.com/zooniverse/mkdocs-deploy-gh-pages` (note: I think I have permission to do this)
2. update the `action.yml` and `Dockerfile` maintainer info to Zooniverse's contact deets.
3. change `uses: mhausenblas/mkdocs-deploy-gh-pages@1.24` to `uses: zooniverse/mkdocs-deploy-gh-pages` (with a version tag, if the fork also copied the version tags)

But that brings to the next point that you mentioned, the _next base image in the chain_ - `squidfunk/mkdocs-material`. What's the recommended plan here?

- (A) also fork `squidfunk/mkdocs-material` to `zooniverse/mkdocs-material` ?
- (B) modify `Dockerfile` in `zooniverse/mkdocs-deploy-gh-pages` to lock the version to `squidfunk/mkdocs-material@8.1.9` ?

Also: where do you usually draw the line for "this is safe enough for our code", and what gotchas should I be aware of (e.g. any FROM dependencies that don't lock down to a version)?

Reply:
> So your recommendation is that we host these GitHub actions on our side, correct? So I'll...

Possibly depending on how much work you want to do / how you feel if you've audited the chain. 

If you are confident in the chain having no issues and lock the version to the one you've audited then i think it's fine to use published actions like you are doing. It's pulling these in without a version lock (and thus may be modified since you looked) that I consider an issue especially if these actions run with access to secrets / access privileges we don't want them to have. 

> But that brings to the next point that you mentioned, the next base image in the chain - squidfunk/mkdocs-material. What's the recommended plan here?

Again - audit the chain and if you are happy then just use it. 

The issue with these GH actions is they can depend on obscured layers (like this one does on `squidfunk/mkdocs-material`) and they become the attack vectors even with the lock version on the action we want to use. This is due to the action rebuilding the image and may pull in new upstream layers depending on the build cache etc. You can't control this sadly :(

However you can vet the chain and make an assessment on the risk, https://github.com/squidfunk

So if you didn't trust this chain at all then the steps you've outlined make sense. You want to recreate the underlying images as one that our org owns (either through forking or code copying) that do the actual work you want the action todo. 

In the past I've forked marketplaces action after vetting to lock it https://github.com/zooniverse/label-when-approved-action and also to combine new features / code into our own versions https://github.com/zooniverse/middleman-gh-pages-action